Antidote/ActiveCallViewController.m

```diff
@@ -234,6 +233,37 @@ - (void)installConstraints
     }];
 }
 
+- (void)viewDidLayoutSubviews
+{
+    [self adjustPreviewLayer];
+}
+
+- (void)updateCallsControlViewExpand
+{
+    self.callControlsView.type = CallsControlsViewTypeExpand;
+
+    [self.callControlsView remakeConstraints:^(MASConstraintMaker *make) {
+        make.top.equalTo(self.topViewContainer.bottom).with.offset(kIndent);
+        make.centerX.equalTo(self.view);
+        make.left.equalTo(self.view).with.offset(kIndent);
+        make.right.equalTo(self.view).with.offset(-kIndent);
+        make.bottom.equalTo(self.view.centerY).with.offset(-kIndent);
+    }];
+
+    self.endCallButton.hidden = NO;
+}
+
+- (void)updateCallsControlViewCondensed
+{
+    self.callControlsView.type = CallsControlsViewTypeCompact;
+    self.endCallButton.hidden = YES;
+
+    [self.callControlsView remakeConstraints:^(MASConstraintMaker *make) {
+        make.bottom.equalTo(self.view);
+        make.left.equalTo(self.view);
+        make.right.equalTo(self.view);
+    }];
+}
```

Comment:
Well yeah, in fact you can add two views to controller - expanded and compact. And then hide/show them in update method.